Update helm templates for compatibility with kustomize

smlx opened the pull request:
**What this PR does / why we need it**:

kustomize now does additional validation on YAML. This PR updates the helm chart templates for compatibility with kustomize.

**Which issue(s) this PR fixes**:
Closes #1022 
Superseded / Closes: #980 

**Special notes for your reviewer**:

The new kustomization patches are based on diffing the generated helm chart to avoid any unrelated changes.

maxsmythe reviewed (approved):
LGTM

Thanks for tackling this!

ritazh commented on cmd/build/helmify/kustomization.yaml:
```diff
@@ -32,3 +32,30 @@ patchesJson6902:
+  # these are defined in the chart values rather than hard-coded
+  - target:
+      kind: Deployment
+      name: gatekeeper-audit
```
Getting the following error:
```
kustomize build cmd/build/helmify | go run cmd/build/helmify/*.go
Error: no matches for OriginalId ~G_~V_Deployment|~X|gatekeeper-audit; no matches for CurrentId ~G_~V_Deployment|~X|gatekeeper-audit; failed to find unique target for patch ~G_~V_Deployment|gatekeeper-audit
```

smlx replied:
Which version of `kustomize` do you have? It works here..
```
$ git describe
v3.3.0-beta.2-15-g2e14e5a3
$ kustomize version
{Version:kustomize/v3.8.8 GitCommit:72262c5e7135045ed51b01e417a7e72f558a22b0 BuildDate:2020-12-10T18:05:35Z GoOs:linux GoArch:amd64}
$ kustomize build cmd/build/helmify > /dev/null; echo $?
0
```

sozercan replied:
I am seeing it in kustomize v3.8.0 but looks like it's fixed in v3.8.8

ritazh replied:
Since this is fixed in v3.8.8, let's update the Makefile to use v3.8.8 instead

sozercan commented:
@smlx there are some conflicts now, can you rebase when you get a chance?

smlx commented:
OK, rebased.

sozercan commented:
LGTM. I think you need to run `make manifests` and commit resulting changes too.

sozercan reviewed (approved):
LGTM

sozercan commented:
@smlx thanks for the PR! Looks like there are a few conflicts, please let us know if you get a chance to resolve them.

smlx commented:
hi @sozercan sorry I've already rebased and fixed conflicts twice now since sending this PR and don't have the time/energy to do it again. Feel free to close or anyone is welcome to take over.

sozercan commented:
No worries at all, thanks for the PR again. I pushed to your branch to fix the conflicts.

@maxsmythe still lgty?

maxsmythe commented on charts/gatekeeper/templates/gatekeeper-controller-manager-deployment.yaml:
```diff
@@ -85,11 +85,11 @@ spec:
           name: healthz
           protocol: TCP
-        readinessProbe:
+        readinessProbe:c
```
there is a stray `c` here

sozercan replied:
good catch! fixed

maxsmythe commented:
LGTM, I think there is one stray character in here though (search for `:c`)

smlx commented:
Thanks for getting this over the line, much appreciated! :tada:
